packages/react/spec/components/SignedInOrRedirect.spec.tsx

```diff
@@ -0,0 +1,59 @@
+import "@testing-library/jest-dom";
+import { render } from "@testing-library/react";
+import React from "react";
+import { superAuthApi } from "../../spec/apis";
+import { TestWrapper } from "../../spec/testWrapper";
+import { expectMockSignedInUser, expectMockSignedOutUser } from "../../spec/utils";
+import { SignedInOrRedirect } from "../../src/components/SignedInOrRedirect";
+
+describe("SignedInOrRedirectOrRedirect", () => {
+  const { location } = window;
+  const mockAssign = jest.fn();
+
+  beforeAll(() => {
+    // @ts-expect-error mock
+    delete window.location;
+    // @ts-expect-error mock
+    window.location = { assign: mockAssign };
+  });
+
+  afterEach(() => {
+    mockAssign.mockClear();
+  });
+
+  afterAll(() => {
+    window.location = location;
+  });
+
+  test("redirects when signed out", () => {
+    const component = (
+      <h1>
+        <SignedInOrRedirect>Hello, Jane!</SignedInOrRedirect>
```

Comment:
I think we're going to need to make the way this thing redirects pluggable, so that when we have our own router, it can push the route client side, or if someone wants to use this with their own router in a next.js project, they can. Thoughts on adding a new provider prop now?